odoo/addons/test_new_api/tests/test_unity_read.py

```diff
@@ -468,3 +472,138 @@ def test_many2many_order_increases_query_count(self):
+            {
+                'id': self.course.id,
+                'reference': {
+                    'id': {'id': self.lesson_day1.id, 'model': self.lesson_day1._name},
+                    'display_name': 'first day'
+                }
+            }
+        ])
+
+    def test_reference_fields_respect_context(self):
+        read = self.course.web_read(
+            {
+                'reference':
+                    {
+                        'fields': {'display_name': {}},
+                        'context': {'special': 'yes'}
+                    }
+            })
+        self.assertEqual(read, [
+            {
+                'id': self.course.id,
+                'reference': {
+                    'id': {'id': self.lesson_day1.id, 'model': self.lesson_day1._name},
+                    'display_name': 'special first day'
+                }
+            }
+        ])
+
+    def test_reference_fields_extra_fields(self):
+        read = self.course.web_read(
+            {
+                'reference':
+                    {
+                        'fields': {'write_date': {}},
+                    }
+            })
+        self.assertEqual(read, [
+            {
+                'id': self.course.id,
+                'reference': {
+                    'id': {'id': self.lesson_day1.id, 'model': self.lesson_day1._name},
+                    'write_date': self.lesson_day1.write_date
+                }
+            }
+        ])
+
+    def test_many2one_reference_naked(self):
+        read = self.course.web_read({'m2o_reference_id': {},
+                                     'm2o_reference_model': {}})
+        self.assertEqual(read, [
+            {
+                'id': self.course.id,
+                'm2o_reference_id': self.lesson_day1.id,
+                'm2o_reference_model': self.lesson_day1._name,
+            }
+        ])
+
+    def test_many2one_reference(self):
+        read = self.course.web_read(
+            {
+                'm2o_reference_id':
+                    {
+                        'fields':
+                            {
+                                'display_name': {},
+                                'write_date': {},
+                            },
+                        'context':
+                            {
+                                'special': 'yes',
+                            }
+                    },
+                'm2o_reference_model': {}
+            })
+        self.assertEqual(read, [
+            {
+                'id': self.course.id,
+                'm2o_reference_id': {
+                    'id': self.lesson_day1.id,
+                    'display_name': "special first day",
+                    'write_date': self.lesson_day1.write_date
+                },
+                'm2o_reference_model': self.lesson_day1._name,
+            }
+        ])
+
+    def test_reference_without_values(self):
+        read = self.course_no_author.web_read(
```

Comment:
You need an extra test where you delete `self.lesson_day1`, then call `self.course.web_read(...)`.